mhw_analysis/ai/ai_mhw_images.py

```diff
@@ -0,0 +1,295 @@
+  #   return None
+  
+  # set up query
+  query = sqlalchemy.select([table]).where(sqlalchemy.and_(
+    # range of lat 
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.lat, lat_start, lat_end
+    ),
+    # and range of lon
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.lon, lon_start, lon_end
+    ),
+    # and range of dates
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.date, datetime.strptime(date_start, '%b %d %Y'), datetime.strptime(date_end, '%b %d %Y')
+    )
+    ))
+  
+  # execute query
+  result = connection.execute(query)
+
+  # get df
+  range_df = DataFrame(result.fetchall())
+  # range_df.columns = query.keys()
+
+  # return df
+  return range_df
+
+# used to save the tuples 
+# could include folder path as a param
+def save_obj(obj, name):
+	# save object to folder
+    with open('mhw_data/'+ name + '.pkl', 'wb') as f:
+        pickle.dump(obj, f, pickle.HIGHEST_PROTOCOL)
+
+def load_obj(name):
+    # load object from folder
+    with open('mhw_data/' + name + '.pkl', 'rb') as f:
+        return pickle.load(f)
+
+def get_mhw_tbl(mhw_file):
+	# create engine and check if it has the table we need
+	engine = sqlalchemy.create_engine('sqlite:///'+mhw_file)
+	engine.has_table("MHW_Events")
+
+	# connect to engine
+	connection = engine.connect()
+
+	# load table from metadata
+	metadata = sqlalchemy.MetaData()
+	mhw_tbl = sqlalchemy.Table('MHW_Events', metadata, autoload=True, autoload_with=engine)
+	
+	# return connection and table
+	return connection, mhw_tbl
+
+def load_z500_noaa(path, date, any_sst):
+	# convert date to proper format
+	ymd = date.year*10000 + date.month*100 + date.day
+
+	# create filepath
+	zfile = os.path.join(path, 'b.e11.B20TRC5CNBDRD.f09_g16.001.cam.h1.Z500.18500101-20051231-043.nc')
+
+	# This is slow, so hold in memory
+	cubes = iris.load(zfile)
+
+	# get cube at ymd date
+	reload(cems_io)
+	z500_cube = cems_io.load_z500(ymd, cubes)
+
+	# get z500 regridded data
+	z500_noaa = z500_cube.regrid(any_sst, iris.analysis.Linear())
+
+	return z500_noaa
+
+def get_noaa_extract(z500_noaa, lat, lon, width):
+	# create boundaries, make separate method when u start iterating over different frames
+	lat_start = (lat-width/2)
+	lat_end = (lat+width/2)
+	lon_start = (lon-width/2)
+	lon_end = (lon+width/2)
+
+	# set boundaries
+	constraint = iris.Constraint(latitude=lambda cell: lat_start < cell < lat_end,
+	                            longitude = lambda cell: lon_start <= cell <= lon_end)
+
+	# get portion of z500 reframe within these boundaries
+	z500_noaa_extract = z500_noaa.extract(constraint)
+
+	return z500_noaa_extract, lat_start, lat_end, lon_start, lon_end
+
+def show_segmentation_map(lats, lons):
+	# ---MAKE PLOTS---
+
+	plt.clf()
+	ax = plt.gca()
+	fig = plt.figure()
+	ax1 = fig.add_subplot(111)
+
+	a = np.stack([np.arange(35.125, 44.875, 0.25) for _ in range(40)])
```

Comment:
Is this just some hard-coded example?

Reply:
yeah that method needs to be revised but it's a task for when I can properly build the dataset.